### PR TITLE
Activate DfcProvider with feature toggle, not prod

### PR DESCRIPTION
**Pull request by mkllnk**
#### What? Why?

I started to work on #9170 and needed to add a feature toggle. That broke several specs and I cleaned it all up to simplify the setup. Hopefully others will find it easier to use feature toggles now.

#### What should we test?

- Check toggling the split checkout on and off.

#### Release notes

Changelog Category: Technical changes

The title of the pull request will be included in the release notes.

### CODE REVIEW

File: spec/controllers/payment_gateways/stripe_controller_spec.rb [hunk 1]
```diff
@@ -88,10 +88,7 @@ module PaymentGateways
 
         context "using split checkout" do
           before do
-            allow(OpenFoodNetwork::FeatureToggle).
-              to receive(:enabled?).with(:split_checkout) { true }
-            allow(OpenFoodNetwork::FeatureToggle).
-              to receive(:enabled?).with(:split_checkout, anything) { true }
+            Flipper.enable(:split_checkout)
```
Comment (jibees) [inline spec/controllers/payment_gateways/stripe_controller_spec.rb:91]: <3

**Review by jibees — APPROVED** [review]
Great, seems perfect to me! 👍

**Comment by jibees** [comment]
One flaky spec: 
https://github.com/openfoodfoundation/openfoodnetwork/issues/9890

**Comment by drummer83** [comment]
Thanks @mkllnk!
Functionality and usability are working as before. I assume the changes you made are only 'behind the scenes', right?
I'll move this to Ready to go! :rocket:

**Pull request merged**